### PR TITLE
Implement split comms watchdog

### PR DESCRIPTION
**Pull request by daskygit**
## Description

Only had minimal testing so far, but this will allow for some feedback.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by KarlK90 — CHANGES_REQUESTED** [review]
Great feature :tada:. From my testing this works great and I found only one tiny edge case.

File: quantum/split_common/split_util.c [hunk 1]
```diff
@@ -74,6 +74,46 @@ static inline bool usbIsActive(void) {
...
+void split_watchdog_init(void) {
+    split_watchdog_started = timer_read32();
+}
+
+void split_watchdog_update(bool done) {
+    split_watchdog_done = done;
+}
+
+bool split_watchdog_check(void) {
+    if (!is_transport_connected()) {
```
[the remark this answers is not in the document]
Comment (daskygit) [reply quantum/split_common/split_util.c:102]: Yeah definitely, I couldn't decide last night if I wanted to require `is_transport_connected` work. I was thinking maybe do a ping on an interval if `SPLIT_MAX_CONNECTION_ERRORS == 0` but think that goes against my intention of not sending data when it's not required.

Anyways, long story short. I'll add an assertion, thanks.

File: quantum/split_common/transactions.c [hunk 2]
```diff
@@ -718,6 +718,36 @@ static void pointing_handlers_slave(matrix_row_t master_matrix[], matrix_row_t s
...
+    bool okay = true;
+    if (!split_watchdog_check()) {
+        okay = transport_write(PUT_WATCHDOG, &okay, sizeof(okay));
+        split_watchdog_update(okay);
+    }
+    return okay;
+}
+
+static void watchdog_handlers_slave(matrix_row_t master_matrix[], matrix_row_t slave_matrix[]) {
+    split_watchdog_update(split_shmem->watchdog_pinged);
```
[the remark this answers is not in the document]
Comment (daskygit) [reply quantum/split_common/transactions.c:736]: I couldn't see a reason to ever require rearming the secondary side watchdog.

**Comment by daskygit** [comment]
Not entirely happy with the DD mapping

File: quantum/split_common/transactions.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (KarlK90) [reply quantum/split_common/transactions.c:736]: Me neither, just double checking the implementation.

**Review by drashna — APPROVED** [review]

**Review by KarlK90 — APPROVED** [review]
LGTM now!

**Review by tzarc — APPROVED** [review]

**Pull request merged**